sparse/unit_test/Test_Sparse_mdf.hpp

```diff
@@ -0,0 +1,149 @@
+  using value_type      = typename crs_matrix_type::value_type;
+
+  constexpr ordinal_type numRows  = 16;
+  constexpr ordinal_type numCols  = 16;
+  constexpr size_type numNonZeros = 64;
+  row_map_type row_map("row map", numRows + 1);
+  col_ind_type col_ind("column indices", numNonZeros);
+  values_type values("values", numNonZeros);
+
+  {  // create matrix
+    const size_type row_mapRaw[]    = {0,  3,  7,  11, 14, 18, 23, 28, 32,
+                                    36, 41, 46, 50, 53, 57, 61, 64};
+    const ordinal_type col_indRaw[] = {
+        0,  1,  4, 0,  1,  2, 5,  1,  2,  3,  6,  2,  3,  7,  0,  4,
+        5,  8,  1, 4,  5,  6, 9,  2,  5,  6,  7,  10, 3,  6,  7,  11,
+        4,  8,  9, 12, 5,  8, 9,  10, 13, 6,  9,  10, 11, 14, 7,  10,
+        11, 15, 8, 12, 13, 9, 12, 13, 14, 10, 13, 14, 15, 11, 14, 15};
+    const value_type values_Raw[] = {
+        4,  -1, -1, -1, 4,  -1, -1, -1, 4,  -1, -1, -1, 4,  -1, -1, 4,
+        -1, -1, -1, -1, 4,  -1, -1, -1, -1, 4,  -1, -1, -1, -1, 4,  -1,
+        -1, 4,  -1, -1, -1, -1, 4,  -1, -1, -1, -1, 4,  -1, -1, -1, -1,
+        4,  -1, -1, 4,  -1, -1, -1, 4,  -1, -1, -1, 4,  -1, -1, -1, 4};
+
+    typename row_map_type::HostMirror::const_type row_map_host(row_mapRaw,
+                                                               numRows + 1);
+    typename col_ind_type::HostMirror::const_type col_ind_host(col_indRaw,
+                                                               numNonZeros);
+    typename values_type::HostMirror::const_type values_host(values_Raw,
+                                                             numNonZeros);
+
+    Kokkos::deep_copy(row_map, row_map_host);
+    Kokkos::deep_copy(col_ind, col_ind_host);
+    Kokkos::deep_copy(values, values_host);
+  }
+
+  crs_matrix_type A = crs_matrix_type("A", numRows, numCols, numNonZeros,
+                                      values, row_map, col_ind);
+
+  KokkosSparse::Experimental::MDF_handle<crs_matrix_type> handle(A);
+  handle.set_verbosity(0);
+  mdf_symbolic_phase(A, handle);
+  mdf_numeric_phase(A, handle);
+
+  col_ind_type permutation = handle.get_permutation();
+
+  bool success = true;
+  typename col_ind_type::HostMirror permutation_h =
+      Kokkos::create_mirror(permutation);
+  Kokkos::deep_copy(permutation_h, permutation);
+  const ordinal_type permutation_ref[] = {0, 3,  12, 15, 1, 2, 4, 8,
+                                          7, 11, 13, 14, 5, 6, 9, 10};
+  printf("MDF ordering: { ");
+  for (ordinal_type idx = 0; idx < A.numRows(); ++idx) {
+    ;
+    printf("%d ", static_cast<int>(permutation_h(idx)));
+    if (permutation_h(idx) != permutation_ref[idx]) {
+      success = false;
+    }
+  }
+  printf("}\n");
+
+  EXPECT_TRUE(success)
+      << "The permutation computed is different from the reference solution!";
+
+  handle.sort_factors();
+  crs_matrix_type U = handle.getU();
+  crs_matrix_type L = handle.getL();
+}
```

Comment:
This is fine for now but eventually it would be good to test the entries and numeric values of L/U by applying U^-1 L^-1 to a vector with trsv - I imagine for this 2D Laplace it would converge really fast?